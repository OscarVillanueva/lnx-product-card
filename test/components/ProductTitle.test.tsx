import React from 'react'
import renderer from 'react-test-renderer'
import { ProductTitle, ProductCard } from '../../src/components';
import products from '../data/products'

describe('ProductTitle Text', () => {
  it('renders without crashing - with custom title', () => {
    
    const wrapper = renderer.create(
      <ProductCard
        product = { products[0] }
      >
        { () => (
          <ProductTitle title='Custom'/>
        )}
      </ProductCard>
    )

    expect(wrapper.toJSON()).toMatchSnapshot()

  });

  test('with product name', () => { 

    const wrapper = renderer.create(
      <ProductCard
        product = { products[0] }
      >
        { () => (
          <ProductTitle />
        )}
      </ProductCard>
    )

    expect(wrapper.toJSON()).toMatchSnapshot()

  })
});
