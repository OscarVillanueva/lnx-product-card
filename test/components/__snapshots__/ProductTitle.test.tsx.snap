// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle Text renders without crashing - with custom title 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  />
</div>
`;

exports[`ProductTitle Text with product name 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - obscuro
  </span>
</div>
`;
